### PR TITLE
[Bugfix] Fix Maverick correctness by filling zero to cache space in cutlass_moe

### PR DESCRIPTION
**Pull request by minosfuture**
## Purpose

#19667 changed the workspace creation from `torch.zeros` to `torch.empty`. This ends up causing correctness for models using cutlass_moe, e.g. Maverick in our test case. This PR fixes the correctness issue by explicitly filling zeros in cutlass_moe.

## Test Plan

lm_eval, ut

## Test Result

lm_eval results:

local-chat-completions (model=meta-llama/Llama-4-Maverick-17B-128E-Instruct-FP8,base_url=http://127.0.0.1:8081/v1/chat/completions,num_concurrent=32), gen_kwargs: (None), limit: 200.0, num_fewshot: 5, batch_size: 1
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value|   |Stderr|
|-----|------:|----------------|-----:|-----------|---|----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.935|±  |0.0175|
|     |       |strict-match    |     5|exact_match|↑  |0.920|±  |0.0192|

unit test stability verified:
* without `c1.fill_(0)`, the following one liner verifies stable failure:
```bash
for i in {1..10}; do echo $i; pytest -s tests/kernels/moe/test_cutlass_moe.py  -k "test_run_cutlass_moe_fp8 or test_cutlass_moe_8_bit_EP_large" -v  2>&1 > /dev/null && { echo "shouldn't succeed"; exit 1; } done`
```
* with `c1.fill_(0)`, the following verifies stable success:
```bash
for i in {1..10}; do echo $i; pytest -s tests/kernels/moe/test_cutlass_moe.py  -k "test_run_cutlass_moe_fp8 or test_cutlass_moe_8_bit_EP_large" -v  2>&1 > /dev/null || { echo "should succeed"; exit 1; } done
```

## (Optional) Documentation Update

**BEFORE SUBMITTING, PLEASE READ <https://docs.vllm.ai/en/latest/contributing>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

File: vllm/model_executor/layers/fused_moe/cutlass_moe.py [hunk 1]
```diff
@@ -176,6 +176,7 @@ def run_cutlass_moe_fp8(
         c1 = _resize_cache(workspace13, (M * topk, N * 2))
         c2 = _resize_cache(workspace2, (M * topk, N))
         c3 = _resize_cache(workspace13, (M * topk, K))
+        c1.fill_(0)
```
Comment (yeqcharlotte) [inline vllm/model_executor/layers/fused_moe/cutlass_moe.py:181]: great! any way to capture this in test_cutlass_moe?

Comment (minosfuture) [reply vllm/model_executor/layers/fused_moe/cutlass_moe.py:181]: yep, added a couple unit tests

File: vllm/model_executor/layers/fused_moe/cutlass_moe.py [hunk 2]
```diff
@@ -177,6 +177,9 @@ def run_cutlass_moe_fp8(
         c2 = _resize_cache(workspace2, (M * topk, N))
         c3 = _resize_cache(workspace13, (M * topk, K))
 
+    if expert_map is not None:
+        c1.fill_(0)
```
Comment (ElizaWszola) [inline vllm/model_executor/layers/fused_moe/cutlass_moe.py:181]: One more tiny thing: can you check if we need to do this if `per_act_token` is true?

Comment (minosfuture) [reply vllm/model_executor/layers/fused_moe/cutlass_moe.py:181]: no we don't. I figured out the root cause is that the random data in the unused space in c1 caused scale (over the whole c1) to be larger, resulting in precision loss for the actual data. So if we use `per_act_token==True`, scales won't be impacted. Let me update the PR in vllm-project.
I'll close this PR to avoid confusion -- this was a experimental PR for early review.

**Comment by minosfuture** [comment]
move to #20167. closing.
